src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -155,12 +155,31 @@
                  (str/join "', '" (sort valid-methods)) "'"))))
   (when (= "regex" (-> rule :match-request :type name str/lower-case))
     (try
-      (re-pattern (:path (:match-request rule)))
+      (let [pattern (re-pattern (:path (:match-request rule)))
+            group-count (-> pattern (.matcher "") .groupCount)
+            get-br-list (fn [match-str]
+                          (map #(Integer/parseInt (nth % 1))
+                               (re-seq #"\$(\d)" match-str)))
+            largest-br (fn [match]
+                         (let [br-list (flatten (map get-br-list
+                                                     (flatten [(or match "")])))]
+                           (when (> (count br-list) 0)
+                             (apply max br-list))))
+            largest-allow-br (or (largest-br (:allow rule)) 0)
+            largest-deny-br (or (largest-br (:deny rule)) 0)]
+        (doseq [[largest field] [[largest-allow-br "allow"]
+                                 [largest-deny-br "deny"]]]
+          (if (> largest group-count)
```

Comment:
Would it be worthwhile to extract into it's own function then? With a docstring explaining what it's doing?

I mean, if the author himself is calling it a pug of an algorithm...are we going to know what the pug it does in 6 months?